src/directives/formly-form.js

```diff
@@ -133,21 +133,58 @@ function formlyForm(formlyUsability, formlyWarn, $parse, formlyConfig, $interpol
 
     function setupFields() {
       $scope.fields = $scope.fields || [];
-      const fieldTransform = $scope.options.fieldTransform || formlyConfig.extras.fieldTransform;
 
-      if (fieldTransform) {
-        $scope.fields = fieldTransform($scope.fields, $scope.model, $scope.options, $scope.form);
-        if (!$scope.fields) {
-          throw formlyUsability.getFormlyError('fieldTransform must return an array of fields');
-        }
+      checkDeprecatedOptions($scope.options);
+
+      let fieldTransforms = $scope.options.fieldTransform || formlyConfig.extras.fieldTransform;
+
+      if (!angular.isArray(fieldTransforms)) {
+        fieldTransforms = [fieldTransforms];
       }
 
+      angular.forEach(fieldTransforms, function transformFields(fieldTransform) {
+        if (fieldTransform) {
+          $scope.fields = fieldTransform($scope.fields, $scope.model, $scope.options, $scope.form);
+          if (!$scope.fields) {
+            throw formlyUsability.getFormlyError('fieldTransform must return an array of fields');
+          }
+        }
+      });
+
       setupModels();
 
       angular.forEach($scope.fields, attachKey); // attaches a key based on the index if a key isn't specified
       angular.forEach($scope.fields, setupWatchers); // setup watchers for all fields
     }
 
+    function warn() {
```

Comment:
Is there a reason you didn't just inject the [`formlyWarn`](https://github.com/formly-js/angular-formly/blob/master/src/services/formlyWarn.js) service and use that instead?